docs/design.md

```diff
@@ -625,21 +625,27 @@ B would require roughly 4G (2\^32 B) to store--too much to duplicate
 between machines. Our conclusion is that range metadata must be
 distributed for large installations.
 
-To distribute the range metadata and keep key lookups relatively fast,
-we use two levels of indirection. All of the range metadata sorts first
-in our key-value map. We accomplish this by prefixing range metadata
-with two null characters (*\0\0*). The *meta1* or *meta2* suffixes are
-additionally appended to distinguish between the first level and second
-level of range metadata. In order to do a lookup for *key1*,
-we first locate the range information for the lower bound of
-`\0\0meta1<key1>`, and then use that range to locate the lower bound
-of `\0\0meta2<key1>`. The range specified there will indicate the
-range location of `<key1>` (refer to examples below). Using two levels
-of indirection, **our map can address approximately 2\^62 B of data, or
-roughly 4E** (*each metadata range addresses 2\^(26-8) = 2\^18 ranges;
-with two levels of indirection, we can address 2\^(18 + 18) = 2\^36
-ranges; each range addresses 2\^26 B; total is 2\^(36+26) B = 2\^62 B =
-4E*).
+To keep key lookups relatively fast in the presence of distributed metadata,
+we store all the top-level metadata in a single range (the first range). These
+top-level metadata keys are known as *meta1* keys, and are prefixed such that
+they sort to the beginning of the key space. Given the metadata size of 256
+bytes given above, a single 64M range would support 64M/256B = 2\^18 ranges,
+which gives a total storage of 64M \* 2\^18 = 16.7T. To support the 1P quoted
+above, we need two levels of indirection, where the first level addresses the
+second, and the second addresses user data. With two levels of indirection, we
+can address 2\^(18 + 18) = 2\^36 ranges; each range addresses 2\^26 B, and
+altogether we address 2\^(36+26) B = 2\^62 B = 4E of user data.
+
+For a given user-addressable `key1`, the associated *meta1* record is found
+at the successor key to `key1` in the *meta1* space. Since the *meta1* space
+is sparse, the successor key is defined as the next key which is present. The
+*meta1* record identifies the range containing the *meta2* record, which is
+found using the same process. The *meta2* record identifies the range
+containing `key1`, which is again found the same way (see examples below).
+
+Concretely, metadata keys are prefixed by `\0\0meta{1,2}`; the two null
+characters provide for the desired sorting behaviour. Thus, `key1`'s
+*meta1* record will reside at the successor key to `\0\0\meta1key1`.
```

Comment:
`s/meta1key1/meta1<key1>/g` to make it clear that the `key1` portion is variable.